packages/core/src/render3/util/discovery_utils.ts

```diff
@@ -24,88 +24,88 @@ import {getTNode, unwrapRNode} from './view_utils';
+ * associated with this DOM element.
+ *
+ * Calling the function on `<my-app>` will return the `MyApp` instance.
+ *
+ *
+ * @param element DOM element from which the component should be retrieved.
+ * @returns Component instance associated with the element or `null` if there
+ *    is no component associated with it.
  *
  * @publicApi
+ * @globalApi ng
  */
-export function getComponent<T = {}>(element: Element): T|null {
-  if (!(element instanceof Node)) throw new Error('Expecting instance of DOM Node');
+export function getComponent<T>(element: Element): T|null {
+  assertDomElement(element);
   const context = loadLContext(element, false);
   if (context === null) return null;
 
-
   if (context.component === undefined) {
     context.component = getComponentAtNodeIndex(context.nodeIndex, context.lView);
   }
 
   return context.component as T;
 }
 
+
 /**
- * Returns the component instance associated with a given DOM host element.
- * Elements which don't represent components return `null`.
- *
- * @param element Host DOM element from which the component should be retrieved.
+ * If inside an embedded view (e.g. `*ngIf` or `*ngFor`), retrieves the context of the embedded
+ * view that the element is part of. Otherwise retrieves the instance of the component whose view
+ * owns the element (in this case, the result is the same as calling `getContainingComponent`).
  *
- * ```
- * <my-app>
- *   #VIEW
- *     <div>
- *       <child-comp></child-comp>
- *     </div>
- * </mp-app>
- *
- * expect(getComponent(<child-comp>) instanceof ChildComponent).toBeTruthy();
- * expect(getComponent(<my-app>) instanceof MyApp).toBeTruthy();
- * ```
+ * @param element Element for which to get the surrounding component instance.
+ * @returns Instance of the component that is around the element or null if the element isn't
+ *    inside any component.
  *
  * @publicApi
+ * @globalApi ng
  */
-export function getContext<T = {}>(element: Element): T|null {
-  if (!(element instanceof Node)) throw new Error('Expecting instance of DOM Node');
+export function getContext<T>(element: Element): T|null {
+  assertDomElement(element);
   const context = loadLContext(element, false);
-  if (context === null) return null;
-
-  return context.lView[CONTEXT] as T;
+  return context === null ? null : context.lView[CONTEXT] as T;
 }
 
 /**
- * Returns the component instance associated with view which owns the DOM element (`null`
- * otherwise).
- *
- * @param element DOM element which is owned by an existing component's view.
+ * Retrieves the component instance whose view contains the DOM element.
  *
+ * @usageNotes
+ * Given the following DOM structure:
  * ```
  * <my-app>
- *   #VIEW
- *     <div>
- *       <child-comp></child-comp>
- *     </div>
- * </mp-app>
- *
- * expect(getViewComponent(<child-comp>) instanceof MyApp).toBeTruthy();
- * expect(getViewComponent(<my-app>)).toEqual(null);
+ *   <div>
```

Comment:
Can we add back the "VIEW" marker here? It's a vital distinction because content children do not work this way. As written, `<child-comp>` would get whatever component `my-app` is inside of.

```suggestion
 * #VIEW
 *   <div>
```

Reply:
@kara the dom snippet is supposed to represent what people see and debug in the browser's elements view. There will be no marker there. How can we make this example more clear while writing it from the perspective of a developer debugging the app?

Reply:
I think in this case having the example from a DOM perspective might be more confusing. Instead, maybe we could clarify with more text?  Something like:

```
For example, if `<child-comp>` is used in the template for `<app-comp>` (i.e. a `ViewChild` of `<app-comp>`), calling `getOwningComponent` on `<child-comp>` would return `<app-comp>`.
```

Reply:
sgtm